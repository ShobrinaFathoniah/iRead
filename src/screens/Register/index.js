import {View, ScrollView} from 'react-native';
import React, {useState} from 'react';
import {
  Forms,
  Header,
  Input,
  LibreBaskerville,
  LoadingBar,
  NoConnection,
} from '../../components';
import {useDispatch, useSelector} from 'react-redux';
import {sendDataRegister} from './redux/action';
import {checkEmail, isValidPassword} from '../../helpers/validationData';
import styles from './style';

const Register = ({navigation}) => {
  const [name, setName] = useState('');
  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');
  const {isLoading, connection} = useSelector(state => state.global);

  const dispatch = useDispatch();

  const dataUser = {
    email: email,
    password: password,
    name: name,
  };

  const register = () => {
    dispatch(sendDataRegister(dataUser, email, password, name, navigation));
  };

  // console.log(email);
  const emailChecker = email => {
    if (!checkEmail(email) && email.length > 0) {
      return (
        <LibreBaskerville style={styles.erorText}>
          Email yang Anda masukan Tidak Valid
        </LibreBaskerville>
      );
    }
  };

  console.log(password);
  const passwordChecker = password => {
    if (!isValidPassword(password) && password.length > 0) {
      return (
        <LibreBaskerville style={[styles.erorText, styles.passwordError]}>
          Password Harus Lebih dari 8 dan Terdapat Angka
        </LibreBaskerville>
      );
    }
  };

  const goToLogin = () => navigation.navigate('Login');

  const registerScreen = () => {
    return (
      <View>
        <Forms type="Register" onPressButton={register} onPressText={goToLogin}>
          <View>
            {emailChecker(email)}
            <Input
              onChangeText={value => setEmail(value)}
              value={email}
              placeholder="Email"
            />
            <Input
              onChangeText={value => setName(value)}
              value={name}
              placeholder="Name"
            />
            {passwordChecker(password)}
            <Input
              onChangeText={value => setPassword(value)}
              value={password}
              placeholder="Password"
              secureTextEntry={true}
            />
          </View>
        </Forms>
        {LoadingBar(isLoading)}
      </View>
    );
  };

  return (
    <ScrollView>
      <Header radiusBottom={true} />
      {connection ? registerScreen() : NoConnection(connection)}
    </ScrollView>
  );
};

export default Register;
