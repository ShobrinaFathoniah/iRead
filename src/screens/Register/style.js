import {StyleSheet} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {LIGHT_BLUE_100_trans, RED_500_trans} from '../../helpers/colors';

const styles = StyleSheet.create({
  erorText: {
    color: RED_500_trans,
    fontSize: moderateScale(12),
    textAlign: 'center',
    padding: moderateScale(10),
    backgroundColor: LIGHT_BLUE_100_trans,
    borderRadius: moderateScale(5),
    width: moderateScale(270),
    alignSelf: 'center',
  },
  passwordError: {
    width: moderateScale(300),
  },
});

export default styles;
