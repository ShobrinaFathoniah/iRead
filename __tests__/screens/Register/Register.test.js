import * as React from 'react';
import Register from '../../../src/screens/Register';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../../src/helpers/reduxTesting';

test('Register Snapshoot', () => {
  const snap = create(ContainerTesting(<Register />));
  expect(snap).toMatchSnapshot();
});
