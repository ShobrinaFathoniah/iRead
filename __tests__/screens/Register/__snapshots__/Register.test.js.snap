// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Snapshoot 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#8180b7",
          "borderBottomEndRadius": 150,
          "borderBottomStartRadius": 150,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "PassionsConflict-Regular",
            },
            Object {
              "color": "#000000",
              "fontSize": 66,
              "letterSpacing": 3.142857142857143,
              "margin": 15.714285714285714,
              "textAlign": "center",
            },
          ]
        }
      >
        iRead
      </Text>
    </View>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Courgette-Regular",
              },
              Object {
                "alignSelf": "center",
                "color": "#272b5a",
                "fontSize": 50.285714285714285,
                "letterSpacing": 0.7857142857142857,
                "marginTop": 23.57142857142857,
              },
            ]
          }
        >
          Register
        </Text>
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/Image/value-is-inside.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 471.42857142857144,
              "resizeMode": "contain",
              "width": 392.85714285714283,
            }
          }
        />
        <View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Email"
              placeholderTextColor="#272b5a"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderBottomWidth": 1.5714285714285714,
                    "borderColor": "#535487",
                    "borderRadius": 7.857142857142857,
                    "color": "#272b5a",
                    "height": 72.28571428571428,
                    "marginBottom": 18.857142857142858,
                    "padding": 15.714285714285714,
                    "width": 424.2857142857143,
                  },
                  Object {},
                ]
              }
              value=""
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Name"
              placeholderTextColor="#272b5a"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderBottomWidth": 1.5714285714285714,
                    "borderColor": "#535487",
                    "borderRadius": 7.857142857142857,
                    "color": "#272b5a",
                    "height": 72.28571428571428,
                    "marginBottom": 18.857142857142858,
                    "padding": 15.714285714285714,
                    "width": 424.2857142857143,
                  },
                  Object {},
                ]
              }
              value=""
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#272b5a"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "borderBottomWidth": 1.5714285714285714,
                    "borderColor": "#535487",
                    "borderRadius": 7.857142857142857,
                    "color": "#272b5a",
                    "height": 72.28571428571428,
                    "marginBottom": 18.857142857142858,
                    "padding": 15.714285714285714,
                    "width": 424.2857142857143,
                  },
                  Object {},
                ]
              }
              value=""
            />
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#535487",
              "borderRadius": 4.714285714285714,
              "margin": 15.714285714285714,
              "opacity": 1,
              "padding": 15.714285714285714,
              "width": 408.57142857142856,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "fontSize": 22,
                },
              ]
            }
          >
            Register
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "margin": 15.714285714285714,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            Have an Account?
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#272b5a",
                    "fontSize": 18.857142857142858,
                    "marginStart": 15.714285714285714,
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
